### DISCUSSION TITLE
Speech-to-text for voice messages

### DISCUSSION THREAD

**Discussion by omyno** [element-hq/element-meta#2572] [Ideas]
Introduction of a (server side? external?) speech-to-text feature for voice messages, similar to what’s available in apps like [Telegram Premium](https://core.telegram.org/api/transcribe). This would generate a written transcript of the voice message.

#### Rationale:

This feature is crucial for accessibility, particularly for hearing-impaired users who may not be able to listen to voice messages.

#### Implementation:

 * A button beside each voice message can trigger the speech-to-text conversion.
 * The transcribed text will appear in a collapsible section within the chat.
 * Additionally, a toggle in settings could enable automatic transcription for all voice messages.

#### Future Enhancements:

In subsequent updates, a local language model (LM) could be integrated to provide automatic summaries of voice messages in addition to transcribing them. This would offer users a concise overview of longer messages, enhancing accessibility and usability.

**Comment by Mushy-Snugglebites-badonkadonk** [comment]
Transcription can happen at the OS level through exposed frameworks and APIs from the manufacturer for on-device transcription. Apple has support for this across iOS and macOS.
